### PR TITLE
[WIP] New package: Slic3rPE-1.42.0

### PR DESCRIPTION
**Pull request by Gigahawk**
Waiting for an official stable release, but I've been using the prebuilt appimage for a while and haven't had any major issues.

Not attempting to build 1.41 since it still has a lot of Perl dependencies that I've had trouble with in the past.

Also I'm renaming the binary from `slic3r-gui` to `slic3r-pe` in case someone wants to get the vanilla Slic3r project building.

### CODE REVIEW

**Comment by Gigahawk** [comment]
Not really sure why builds are failing on travis, they seem to be working fine locally. Except for aarch64 builds, those get all the way to the end and then errors out when stripping the binary

File: srcpkgs/Slic3rPE/template [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# Template file for 'Slic3rPE'
+pkgname=Slic3rPE
+version=1.42.0
+revision=1
+_status="-alpha6"
+wrksrc="Slic3r-version_${version}${_status}"
+build_style=cmake
+configure_args="-DSLIC3R_WX_STABLE=1 -DSLIC3R_FHS=1 -DDESTDIR=/usr/share/"
+hostmakedepends="pkg-config"
+makedepends="boost-devel tbb-devel libcurl-devel nlopt-devel gtest-devel wxWidgets-devel glu-devel libpng-devel eigen glew-devel"
```
Comment (maxice8) [inline srcpkgs/Slic3rPE/template]: 80 columns

File: srcpkgs/Slic3rPE/template [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+depends="boost tbb libcurl nlopt-devel wxWidgets glu libpng eigen glew"
```
Comment (maxice8) [inline srcpkgs/Slic3rPE/template]: aren't they automatically found via shlibs ?

File: srcpkgs/Slic3rPE/template [hunk 3]
```diff
@@ -0,0 +1,21 @@
+# Template file for 'Slic3rPE'
+pkgname=Slic3rPE
+version=1.42.0
+revision=1
+_status="-alpha6"
```
Comment (maxice8) [inline srcpkgs/Slic3rPE/template]: alpha software ?

Comment (Gigahawk) [reply srcpkgs/Slic3rPE/template]: Please see the original description I wrote

**Comment by maxice8** [comment]
Staill failing on native x86_64-musl

**Comment by maxice8** [comment]
Staill failing on native x86_64-musl

**Comment by Johnnynator** [comment]
I'm working on the musl issue (The issue is lying in boost not this package). The static_assert in their header is wrong for musl.

**Comment by Johnnynator** [comment]
See #9595 for boost
Needed patch for musl
```diff
--- src/png/error.hpp	2019-03-01 18:52:12.000000000 +0100
+++ -	2019-03-10 03:20:30.151491676 +0100
@@ -100,7 +100,7 @@
             strerror_s(buf, ERRBUF_SIZE, errnum);
             return std::string(buf);
 #else
-#if (_POSIX_C_SOURCE >= 200112L || _XOPEN_SOURCE >= 600) && !_GNU_SOURCE
+#if (_POSIX_C_SOURCE >= 200112L || _XOPEN_SOURCE >= 600) && !_GNU_SOURCE || !defined(__GLIBC__)
             char buf[ERRBUF_SIZE] = { 0 };
             strerror_r(errnum, buf, ERRBUF_SIZE);
             return std::string(buf);
--- src/slic3r/Utils/Serial.cpp	2019-03-01 18:52:12.000000000 +0100
+++ -	2019-03-10 03:25:43.045707061 +0100
@@ -42,8 +42,8 @@
 #ifndef _WIN32
 	#include <sys/ioctl.h>
 	#include <sys/time.h>
-	#include <sys/unistd.h>
 	#include <sys/select.h>
+	#include <unistd.h>
 #endif
 
 #if defined(__APPLE__) || defined(__OpenBSD__)
```

**Pull request closed without merging**